pinecone/provider/index_resource.go

```diff
@@ -266,18 +419,72 @@ func (r *IndexResource) Create(ctx context.Context, req resource.CreateRequest,
 	}
 
 	if spec.Serverless != nil {
-		serverlessReq := pinecone.CreateServerlessIndexRequest{
-			Name:      data.Name.ValueString(),
-			Dimension: int32(data.Dimension.ValueInt64()),
-			Metric:    pinecone.IndexMetric(data.Metric.ValueString()),
-			Cloud:     pinecone.Cloud(spec.Serverless.Cloud.ValueString()),
-			Region:    spec.Serverless.Region.ValueString(),
-		}
+		metric := pinecone.IndexMetric(*data.Metric.ValueStringPointer())
+		deletionProtection := pinecone.DeletionProtection(data.DeletionProtection.ValueString())
 
-		_, err := r.client.CreateServerlessIndex(ctx, &serverlessReq)
-		if err != nil {
-			resp.Diagnostics.AddError("Failed to create serverless index", err.Error())
-			return
+		if embed != nil {
+			if spec.Pod != nil {
+				resp.Diagnostics.AddError("Invalid configuration", "Integrated indexes must have a serverless spec.")
+				return
+			}
+
+			fieldMap := mapAttrToInterfacePtr(embed.FieldMap)
+			if fieldMap == nil {
+				resp.Diagnostics.AddError("Invalid configuration", "Integrated indexes must have a field_map")
```

Comment:
It's nice people will see very clear and specific error messages. But it seems a shame to have all this validation baked into the provider if any of these limitations change in the future.